zebra-chain/src/history_tree.rs

```diff
@@ -97,33 +180,38 @@ impl HistoryTree {
                 height, self.current_height
             );
         }
+        let network_upgrade = NetworkUpgrade::current(self.network, height);
+        if network_upgrade != self.network_upgrade {
+            // This is the activation block of a network upgrade.
+            // Create a new tree.
+            let new_tree = Self::from_block(self.network, block, sapling_root, orchard_root)?;
+            *self = new_tree;
```

Comment:
👀

Reply:
Probably worth documenting that the method replaces `&mut self`.